client/cli/src/lib.rs

```diff
@@ -19,127 +19,34 @@
 #![warn(missing_docs)]
 #![warn(unused_extern_crates)]
 
-#[macro_use]
-mod traits;
 mod params;
-mod execution_strategy;
-pub mod error;
+mod arg_enums;
+mod error;
 mod runtime;
 mod node_key;
+mod commands;
 
-use sc_client_api::execution_extensions::ExecutionStrategies;
-use sc_service::{
-	config::{Configuration, DatabaseConfig, KeystoreConfig},
-	ServiceBuilderCommand,
-	RuntimeGenesis, ChainSpecExtension, PruningMode, ChainSpec,
-	AbstractService, Roles as ServiceRoles,
-};
 pub use sc_service::config::VersionInfo;
-use sc_network::{
-	self,
-	multiaddr::Protocol,
-	config::{
-		NetworkConfiguration, TransportConfig, NonReservedPeerMode,
-	},
-};
-
-use std::{
-	io::Write, iter, fmt::Debug, fs,
-	net::{Ipv4Addr, SocketAddr}, path::PathBuf,
-};
+
+use std::io::Write;
 
 use regex::Regex;
-use structopt::{StructOpt, clap};
+use structopt::{StructOpt, clap::{self, AppSettings}};
 pub use structopt;
-use params::{
-	NetworkConfigurationParams, TransactionPoolParams, Cors,
-};
-pub use params::{
-	SharedParams, ImportParams, ExecutionStrategy, Subcommand, RunCmd, BuildSpecCmd,
-	ExportBlocksCmd, ImportBlocksCmd, CheckBlockCmd, PurgeChainCmd, RevertCmd,
-	BenchmarkCmd,
-};
-pub use traits::GetSharedParams;
-use app_dirs::{AppInfo, AppDataType};
+pub use params::*;
+pub use commands::*;
+pub use arg_enums::*;
+pub use error::*;
```

Comment:
@bkchr a lot more things are now directly accessible without having to go into the submodules. But a lot more things is also exposed (more than before). For example, all the arg_enums are now exposed. This allow another crate to manipulate them if they want (or re-use them). Anyway, all structopt should be recursively public I think